### PR TITLE
Add a Memory Session storage to store sessions

### PR DESCRIPTION
**Pull request by idjaw**
This will improve the way netman store his sessions and this will allow extendability.

### CODE REVIEW

File: netman/core/switch_sessions.py [hunk 1]
```diff
@@ -15,13 +15,16 @@
 import logging
 import threading
 
+from netman.adapters.memory_session_storage import MemorySessionStorage
 from netman.core.objects.exceptions import UnknownSession, SessionAlreadyExists
 
 
 class SwitchSessionManager(object):
-    def __init__(self, session_inactivity_timeout=60):
+    def __init__(self, session_storage=None, session_inactivity_timeout=60):
```
Comment (lindycoder) [inline netman/core/switch_sessions.py]: When adding a new parameter add it at the end so you keep the backward compatibility if people was using SwitchSessionManager(30)

File: netman/core/switch_sessions.py [hunk 2]
```diff
@@ -15,13 +15,16 @@
...
+        if not session_storage:
+            session_storage = MemorySessionStorage()
+        self.session_storage = session_storage
```
Comment (lindycoder) [inline netman/core/switch_sessions.py]: could be
```
self.session_storage = session_storage or MemorySessionStorage()
```

Comment (fbouliane) [inline netman/core/switch_sessions.py]: Sweet !

**Comment by idjaw** [comment]
After discussions, we will be re-working the architecture of this. Closing for now and re-working with new architecture layout.

**Pull request closed without merging**